pandas/core/indexes/numeric.py

```diff
@@ -121,6 +121,8 @@ def _validate_fill_value(self, value):
             # force conversion to object
             # so we don't lose the bools
             raise TypeError
+        if isinstance(value, str):
```

Comment:
could be elif (also shouldn't we be validating instead that this is a float/int) e.g. a datetime would pass right thru here

Reply:
yah, the way we use this method is weird.  i think im going to have to get our usage to be not-weird before i can fully clean up this method